core/variant/dictionary.cpp

```diff
@@ -313,6 +351,146 @@ Array Dictionary::values() const {
 	return varr;
 }
 
+void Dictionary::assign(const Dictionary &p_dictionary) {
+	const ContainerTypeValidate &typed_key = _p->typed_key;
+	const ContainerTypeValidate &typed_key_source = p_dictionary._p->typed_key;
+
+	const ContainerTypeValidate &typed_value = _p->typed_value;
+	const ContainerTypeValidate &typed_value_source = p_dictionary._p->typed_value;
+
+	if ((typed_key == typed_key_source || typed_key.type == Variant::NIL || (typed_key_source.type == Variant::OBJECT && typed_key.can_reference(typed_key_source))) &&
+			(typed_value == typed_value_source || typed_value.type == Variant::NIL || (typed_value_source.type == Variant::OBJECT && typed_value.can_reference(typed_value_source)))) {
+		// from same to same or
+		// from anything to variants or
+		// from subclasses to base classes
```

Comment:
```suggestion
		// From same to same or,
		// from anything to variants or,
		// from subclasses to base classes.
```
[Comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide) nitpick